src/main/java/tiger/command/Command.java

```diff
@@ -0,0 +1,118 @@
+            AppState newApplicationState;
+            switch (Parser.getCommandKeyword(command)) {
+            case "y":
+                newApplicationState = new AppState(applicationState.getTaskList(),
+                        Flag.STORAGE_PARTIAL_LOAD);
+                return new StorageLoadAction(newApplicationState);
+            case "n":
+                newApplicationState = new AppState(applicationState.getTaskList(),
+                        Flag.STORAGE_WIPE);
+                return new StorageLoadAction(newApplicationState);
+            default:
+                newApplicationState = new AppState(applicationState.getTaskList(),
+                        Flag.STORAGE_FAILED);
+                return new StorageLoadAction(newApplicationState);
+            }
+        }
+        switch (Parser.getCommandKeyword(command)) {
+        case "exit":
+            // Fallthrough
+        case "quit":
+            // Fallthrough
+        case "bye":
+            return new ByeAction(applicationState);
+        case "ls":
+            // Fallthrough
+        case "list":
+            return new ListAction(applicationState);
+        case "done":
+            MarkDoneParser markDoneParser = new MarkDoneParser(command);
+            markDoneParser.parse();
+            return new MarkDoneAction(applicationState, markDoneParser.getIndex() - 1);
+        case "remove":
+            // Fallthrough
+        case "del":
+            // Fallthrough
+        case "delete":
+            DeleteParser deleteCommand = new DeleteParser(command);
+            deleteCommand.parse();
+            return new DeleteAction(applicationState, deleteCommand.getIndex() - 1);
+        case "todo":
+            ToDoParser toDoCommand = new ToDoParser(command);
+            toDoCommand.parse();
+            return new ToDoAction(applicationState, toDoCommand.getTodo(), toDoCommand.getPriority());
+        case "dateline":
+            // Fallthrough;
+        case "deadline":
+            DeadLineParser deadLineCommand = new DeadLineParser(command);
+            deadLineCommand.parse();
+            return new DeadLineAction(applicationState, deadLineCommand.getTodo(), deadLineCommand.getDate(),
+                    deadLineCommand.getPriority());
+        case "event":
+            EventParser eventCommand = new EventParser(command);
+            eventCommand.parse();
+            return new EventAction(applicationState, eventCommand.getTodo(), eventCommand.getDate(),
+                    eventCommand.getPriority());
+        case "search":
+            // Fallthrough
+        case "find":
+            FindParser findCommand = new FindParser(command);
+            findCommand.parse();
+            return new FindAction(applicationState, findCommand.getToSearchFor());
+        case "clear":
+            return new ClearAction(applicationState);
+        default:
+            return new InvalidAction(applicationState);
+        }
+    }
```

Comment:
like how you have considered so many scenarios

Reply:
Thank you for the kind words!